src/List.elm

```diff
@@ -244,18 +238,20 @@ product numbers =
 
 {-| Find the maximum element in a non-empty list.
 
-    maximum [1,4,2] == 4
+    maximum 0 [1,4,2] == Just 4
+    maximum 0 []      == 0
 -}
-maximum : List comparable -> comparable
-maximum = foldl1 max
+maximum : comparable -> List comparable -> comparable
+maximum def list = foldl max def list
 
 
 {-| Find the minimum element in a non-empty list.
 
-    minimum [3,2,1] == 1
+    minimum 10 [3,2,1] == Just 1
+    minimum 10 []      == 10
 -}
-minimum : List comparable -> comparable
-minimum = foldl1 min
+minimum : comparable -> List comparable -> comparable
+minimum def list = foldl min def list
```

Comment:
I think it is often hard to pick a minimum value. I'm imagining people writing `minimum (maximum 0 xs) xs`.

Why did you go for this instead of returning a maybe?